discretisedfield/field.py

```diff
@@ -3475,3 +3476,115 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
         else:
             return self.__class__(mesh[0], dim=result.shape[-1], value=result,
                                   components=self.components)
+
+    def to_xarray(self, name='field', units=None):
+        """Field value as ``xarray.DataArray``.
+
+        The function returns an ``xarray.DataArray`` with dimensions ``x``,
+        ``y``, ``z``, and ``comp`` (``only if field.dim > 1``). The coordinates
+        of the geometric dimensions are derived from ``self.mesh.axis_points``,
+        and for vector field components from ``self.components``. The ``units``
+        attribute of geometric dimensions is set to
+        ``self.mesh.attributes['unit']``.
+
+        The name and units of the field ``DataArray`` can be set by passing
+        ``name`` and ``units``. If the type of value passed to any of the two
+        arguments is not ``str``, then a ``TypeError`` is raised.
+
+        Parameters
+        ----------
+        name : str, optional
+
+            String to set name of the field ``DataArray``.
+
+        units : str, optional
+
+            String to set units of the field ``DataArray``.
+
+        Returns
+        -------
+        xarray.DataArray
+
+            Field values DataArray.
+
+        Raises
+        ------
+        TypeError
+
+            If either ``name`` or ``units`` argument is not a string.
+
+        Examples
+        --------
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (10, 10, 10)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh=mesh, dim=3, value=(1, 0, 0), norm=1.)
+        ...
+        >>> field
+        Field(...)
+        >>> field.to_xarray()
```

Comment:
We could have additional "headings" (the snippet isn't properly highlighted on GitHub, headings are `rst`):

```python
1. [Create a field]
>>> import discretisedfield as df
[...]
2. [Create `xarray.DataArray`]
>>> xa = field.to_xarray()
>>> xa
[...]
3. Extract a single point
>>> xa.sel(...)
[...]
[and so on]
```
(see also my comment below)

Reply:
Good idea! I have added headings.